microsoft/testsuites/cloud_hypervisor/ch_tests_tool.py

```diff
@@ -250,49 +250,39 @@ def _extract_test_results(
         # { "type": "test", "event": "ok", "name": "integration::test_vfio" }
         lines = output.split("\n")
         for line in lines:
-            matches = re.findall(r"{.*}", line)
-            if not matches:
+            if '"type": "test"' not in line:
                 continue
-            line = matches[0]
-            result = {}
-            try:
-                result = json.loads(line)
-            except json.decoder.JSONDecodeError:
-                continue
-
-            if type(result) is not dict:
-                continue
-
-            if "type" not in result or result["type"] != "test":
-                continue
-
-            if "event" not in result or result["event"] not in [
-                "started",
-                "ok",
-                "failed",
-                "ignored",
-            ]:
-                continue
-
-            if result["event"] == "started":
-                status = TestStatus.RUNNING
-            elif result["event"] == "ok":
-                status = TestStatus.PASSED
-            elif result["event"] == "failed":
-                status = TestStatus.FAILED
-            elif result["event"] == "ignored":
-                status = TestStatus.SKIPPED
+            for result in extract_jsons(line):
```

Comment:
We discussed that we would first try to parse the line as-is and if that fails, we will try `extract_jsons`. Why did you deviate from that?

Reply:
Changed it to parse the line as-is initially and try `extract_jsons` for failing case.